python/cudf/bindings/join.pyx

```diff
@@ -88,10 +89,10 @@ cpdef join(col_lhs, col_rhs, on, how, method='sort'):
         if how == 'left':
             result = gdf_left_join(list_lhs,
                 col_lhs_len,
-                <int*>left_idx.data,
+                &left_idx[0],
```

Comment:
Will `gdf_*_join` perform the proper bounds-checks with this pointer and the length(s)?

Reply:
AFAIK its up to the C++ side, the cython boundscheck is just for the cython side of the code (i.e. if you were doing the computations with cython code), C++ can still out of bound with this side of bounds check. Might create a quick test of that in a little bit if this PR is still open by then...

Reply:
It makes sense to me to have bounds checks for both Cython and C++. These checks create more visibility into errors that crop up when calls to this binding are made. Not sure if this belongs here, though.

Reply:
Given it's a warning and we do pretty extensive bounds checking in the functions an end user would call above this I'm going to close this for now, but we can revisit if it's an issue later.